packages/frontend/src/Build.svelte

```diff
@@ -23,28 +26,46 @@ let buildFolder: string;
 let buildType: BuildType[];
 let buildArch: string | undefined;
 
-// Other variabler
+// Other variable
 let success = false;
 let buildInProgress = false;
-let bootcAvailableImages: any[] = [];
+let bootcAvailableImages: ImageInfo[] = [];
 let errorMessage = '';
 let errorFormValidation = '';
 
+function findImage(repoTag: string): ImageInfo | undefined {
+  return bootcAvailableImages.find(
+    image => image.RepoTags && image.RepoTags.length > 0 && image.RepoTags[0] === repoTag,
+  );
+}
 // Function that will use listHistoryInfo, if there is anything in the list, pick the first one in the list (as it's the most recent)
 // and fill buildFolder, buildType and buildArch with the values from the selected image.
 async function fillBuildOptions() {
+  // Fill the build options from history
   const historyInfo = await bootcClient.listHistoryInfo();
   if (historyInfo.length > 0) {
     const latestBuild = historyInfo[0];
     buildFolder = latestBuild.folder;
     buildType = latestBuild.type;
     buildArch = latestBuild.arch;
+  }
 
-    // Find the image that matches the latest build's name and tag and set selectedImage to that value
-    selectedImage = bootcAvailableImages.find(
-      image =>
-        image.RepoTags && image.RepoTags.length > 0 && image.RepoTags[0] === `${latestBuild.image}:${latestBuild.tag}`,
-    );
+  // If an image name and tag were passed in, try to use it as the initially selected image
+  let initialImage: ImageInfo | undefined;
+  if (imageName && imageTag) {
+    console.log('Preselecting image: ' + imageName + ' ' + imageTag);
+    initialImage = findImage(`${imageName}:${imageTag}`);
+  }
+
+  // If not, use the last image from history if it is valid
+  if (!initialImage && historyInfo.length > 0) {
+    const latestBuild = historyInfo[0];
+    // Find the image that matches the latest build's name and tag
+    initialImage = findImage(`${latestBuild.image}:${latestBuild.tag}`);
```

Comment:
should we check if .image + .tag isn't blank / undefined?

Reply:
Well, we weren't before and it'll just fail silently :) but I check and simplified this in next commit.